### ISSUE TITLE
Pin a particular workflow run to prevent it from getting purged

### ISSUE THREAD

**Issue by utkarshpaliwal9** [temporalio/temporal#3863]
**Problem Description**
Coming from the fact there are some unusual/rare events that happen in a run (like some race condition, for example, which are difficult to reproduce) that needs to be looked into at a later point in time. And the devs have to race against time to solve the issue before the run gets purged.

**Describe the solution you'd like**
Temporal can provide a button to mark/pin/save a run and this makes sure the run would not get purged. Later, this workflow run can be referenced with a link or otherwise. This run stored in the DB should not get deleted unless explicitly deleted.

**Alternatives considered**
Downloading workflow history json probably would work as a stopgap for saving it locally. But, some better way from the Temporal UI would be really helpful such that others can also look at it.

**Comment by yiminc** [comment]
This is for closed workflow? I think you are asking for ability to dynamic retention per workflow.

**Comment by yiminc** [comment]
This is currently not possible, could be a future feature to consider.

**Comment by utkarshpaliwal9** [comment]
Yes @yiminc. This feature ask indeed is for a closed workflow to prevent it from getting purged eventually.
